cmd/utils/flags.go

```diff
@@ -1652,8 +1668,18 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
 	if ctx.GlobalIsSet(TriesInMemoryFlag.Name) {
 		cfg.TriesInMemory = ctx.GlobalUint64(TriesInMemoryFlag.Name)
 	}
-	if ctx.GlobalIsSet(AllowInsecureNoTriesFlag.Name) {
-		cfg.NoTries = ctx.GlobalBool(AllowInsecureNoTriesFlag.Name)
+	if ctx.GlobalIsSet(TriesVerifyModeFlag.Name) {
+		cfg.TriesVerifyMode = *GlobalTextMarshaler(ctx, TriesVerifyModeFlag.Name).(*core.VerifyMode)
+		// If a node sets verify mode to full or light, it's a fast node and need
+		// to verify blocks from verify nodes, then it should enable trust protocol.
+		if cfg.TriesVerifyMode.NeedRemoteVerify() {
+			cfg.EnableTrustProtocol = true
+		}
+		// If a node sets verify mode but not local, it's a fast node whose difflayer is not integral.
+		// So fast node should disable diff protocol.
+		if cfg.TriesVerifyMode != core.LocalVerify {
```

Comment:
After discuss with nash, we can delete flag `DisableDiffProtocol`, even fast node can provide difflayer for diff sync.